classes/class-wc-connect-paypal-ec.php

```diff
@@ -0,0 +1,250 @@
+					empty( $settings->sandbox_api_subject ) &&
+					empty( $settings->sandbox_api_username ) &&
+					empty( $settings->api_username )
+				) {
+					$email = isset( $settings->email ) ? $settings->email : $settings->api_subject;
+					$settings->api_subject = $email;
+					$settings->sandbox_api_subject = $email;
+					$settings->save();
+				}
+
+				$username = $settings->get_active_api_credentials()->get_username();
+				$subject  = $settings->get_active_api_credentials()->get_subject();
+
+				if ( empty( $username ) && ! empty( $subject ) ) {
+					// Reroute Express Checkout requests from the PPEC extension via WCS to pick up API credentials
+					add_filter( 'woocommerce_paypal_express_checkout_request_body', array( $this, 'request_body' ) );
+
+					add_filter( 'option_woocommerce_ppec_paypal_settings', array( $this, 'paypal_ec_settings' ) );
+					add_filter( 'woocommerce_payment_gateway_supports', array( $this, 'supports' ), 10, 3 );
+
+					add_filter( 'pre_option_wc_gateway_ppce_prompt_to_connect', '__return_empty_string' );
+					if ( 'sandbox' !== $settings->environment ) {
+						add_action( 'woocommerce_order_status_on-hold', array( $this, 'maybe_set_banner' ) );
+						add_action( 'woocommerce_payment_complete', array( $this, 'maybe_set_banner' ) );
+						add_action( 'current_screen', array( $this, 'maybe_init_banner' ) );
+					}
+				}
+			}
+		}
+
+		public function request_body( $body ) {
+			$methods_to_proxy = array( 'SetExpressCheckout', 'GetExpressCheckoutDetails', 'DoExpressCheckoutPayment' );
+			if ( in_array( $body['METHOD'], $methods_to_proxy ) ) {
+				add_filter( 'woocommerce_paypal_express_checkout_request_endpoint', array( $this, 'request_endpoint' ), 10, 2 );
+			} else {
+				remove_filter( 'woocommerce_paypal_express_checkout_request_endpoint', array( $this, 'request_endpoint' ), 10, 2 );
+			}
+			return $body;
+		}
+
+		/**
+		 * Get WCS PayPal proxy endpoint
+		 */
+		public function request_endpoint( $endpoint, $environment ) {
+			return trailingslashit( WOOCOMMERCE_CONNECT_SERVER_URL ) . 'paypal/nvp/' . $environment;
+		}
+
+		/**
+		 * Limit supported payment gateway features to payments
+		 */
+		public function supports( $supported, $feature, $gateway ) {
+			return 'ppec_paypal' === $gateway->id ? 'products' === $feature : $supported;
+		}
+
+		public function maybe_set_banner( $order_id ) {
+			$order          = wc_get_order( $order_id );
+			$payment_method = WC_Connect_Compatibility::instance()->get_payment_method( $order );
+
+			if ( 'ppec_paypal' === $payment_method ) {
+				update_option( 'wc_connect_banner_ppec', 'yes' );
+			}
+		}
+
+		/**
+		 * Once a payment is received, show prompt to connect a PayPal account on certain screens
+		 */
+		public function maybe_init_banner() {
+			if (
+				'yes' !== get_option( 'wc_connect_banner_ppec', null ) ||
+				'yes' === get_option( 'wc_connect_dismiss_banner_ppec', null )
+			) {
+				return;
+			}
+
+			$screen = get_current_screen();
```

Comment:
Now that this function is attached to `current_screen`, you can just access the screen object passed in.